### ISSUE TITLE
Extra text can be entered into checkbox cells of modify table field grid

### ISSUE THREAD

**Issue by chrisjlocke** [sqlitebrowser/sqlitebrowser#598]
I'm using Windows OS and Qt 5.4.1 (but also happens in Qt 4.8.7 ... laptop here has 5.4.1 but work PC which has a newer build of SB has an older Qt...)
When modifying a table, there are checkboxes for primary key, not null, etc.  If these are double-clicked, text can be entered into the cell.
![image](https://cloud.githubusercontent.com/assets/7872218/15438228/09ab1a62-1ec3-11e6-98de-8685ce4c5b3c.png)

Obviously its only a cosmetic issue as the text is ignored, and I presume you don't want to bloat the code with 'cleanup' code on the off-chance some muppet will enter text, but didn't know if the table cell allowed editing, but disabled text entering for particular columns.

**Comment by justinclift** [comment]
Reading through that, I'm a bit uncertain about it.  When you say "text can be entered into the cell", you're meaning the "a", "b", "c", d" above in the Not NULL, PK, AI, and U fields?

If so... that does sound like a bug.  Not sure if it's in DB4S or in Qt though.  If it's in our code, we should fix it. :smile:

**Comment by chrisjlocke** [comment]
Yes, the 'a', 'b', 'c' and 'd' can be entered by double clicking in the cell where the checkbox is.  Interestingly, when editing an existing table, because its a change in the row, the usual 'modify table' SQL process is run (but obviously ignored).
In the 'real world', I can't see anyone double-clicking in the cells, and if they did, they wouldn't enter text...
I presume its because the cell is empty and can contain checkboxes, rather than being solely a checkbox.  Again, on my Windows version, the column can be resized out, but cannot be resized back down to the original size ... as if it 'should' contain text...

**Comment by chrisjlocke** [comment]
No-one would do it + and it has no bearing on the program + and entirely cosmetic = waste of time!

**Comment by justinclift** [comment]
Errr, I've been meaning to look into this one, even if just for improving my understanding of Qt widgets. :smile:

**Comment by innermous** [comment]
I think this class is a bit messy. It requires a bit MVC rework imo. I'll send a patch soon and we can discuss if it's better than current implementation.

**Comment by mgrojo** [comment]
This should be fixed in the next nightly, together with other minor improvements in the Edit Table dialog.
